lib/puppet/provider/package/gem.rb

```diff
@@ -46,6 +50,18 @@ def self.provider_command
     is_optional
   end
 
+  # Having puppet/bin in PATH makes gem provider to use puppet/bin/gem
+  # This is an utiliy methods that reads the PATH and returns a string
+  # that contains the content of PATH but without puppet/bin dir.
+  # This is used to pass a custom PATH as a specific environment for executed commands
+  def self.windows_path_without_puppet_bin
+    @path ||= Puppet::Util.get_env('PATH').split(File::PATH_SEPARATOR)
+                                          .reject { |dir| dir =~ /puppet\\bin/ }
```

Comment:
Does puppet's bin directory always contain backslashes in the `PATH` environment variable? Maybe use `Pathname#cleanpath` to normalize?

```
irb(main):002:0> Pathname.new("C:\\Puppet").cleanpath
=> #<Pathname:C:/Puppet>
```

Should we only reject entries that end with `/puppet\\bin$/`?

Reply:
> Does puppet's bin directory always contain backslashes in the PATH environment variable?

I believe so, since this is controlled in environment.rb(https://github.com/puppetlabs/puppet-agent/blob/master/resources/files/windows/environment.bat#L13) and we also use backslashes in daemon:(https://github.com/puppetlabs/puppet/blob/main/ext/windows/service/daemon.rb#L203-L206)

> Should we only reject entries that end with /puppet\\bin$/?

Yes